pkg/workflows/upgrade.go

```diff
@@ -207,6 +210,17 @@ func (s *upgradeCoreComponents) Run(ctx context.Context, commandContext *task.Co
 	}
 	commandContext.UpgradeChangeDiff.Append(changeDiff)
 
+	target := commandContext.ManagementCluster
+	if commandContext.ManagementCluster.ExistingManagement {
```

Comment:
Good catch. It still worked since the management cluster already had the release CRD, but it shouldn't apply the release CRD to the workload cluster. Fixed it!